src/machine/CoffeeMachine.java

```diff
@@ -0,0 +1,80 @@
+package machine;
+
+class CoffeeMachine {
+
+    private final Supplies supplies;
+
+    // For our purposes it is sufficient that the price has int type
+    private int money;
+
+    private CoffeeMachineState currentState;
+
+    CoffeeMachineState getCurrentState() {
+        return currentState;
+    }
+
+    CoffeeMachine(Supplies supplies, int money) {
+        this.supplies = supplies;
+        this.money = money;
+        this.currentState = CoffeeMachineState.CHOOSING_AN_ACTION;
+    }
+
+    private void displayContent() {
+        System.out.println("\nThe coffee machine has:");
+        supplies.displaySupplies();
+        System.out.printf("$%d of money\n", money);
+    }
+
+    private void prepareCoffee(int chosenOption) {
+        final var coffee = Coffee.fromOption(chosenOption);
+        if (coffee == null || !isEnoughIngredients(coffee)) return;
+        supplies.reduceSupplies(coffee);
+        money += coffee.getPriceInDollars();
+    }
+
+    private void takeMoney() {
```

Comment:
minor: let's highlight in the name that we take all money from the coffee machine

Reply:
Would `takeAllMoney` be a better name?